src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisResultBuilder.cs

```diff
@@ -49,6 +53,37 @@ internal AnalysisResultBuilder(bool logAnalyzerExecutionTime, ImmutableArray<Dia
             return map;
         }
 
+        private static ImmutableDictionary<string, OneOrMany<AdditionalText>> CreatePathToAdditionalTextMap(ImmutableArray<AdditionalText> nonSourceFiles)
+        {
+            if (nonSourceFiles.IsEmpty)
+            {
+                return ImmutableDictionary<string, OneOrMany<AdditionalText>>.Empty;
```

Comment:
This needs to use `PathUtilities.Comparer`

Reply:
Even for empty dictionary?

Reply:
Yes. Let's use the following smaller example to demonstrate the problem:

```c#
ImmutableDictionary<string, Widget> GetMap() {
  if (someCondition) { 
    return ImmutableDictionary<string, Widget>.Empty;
  }

  return new ImmutableDictionary<string, Widget>(PathUtilities.Comparer, ...);
}
```

Consider the impact on the caller when they continue to evolve the returned value:

```c#
var map = GetMap()
  .Add("Example", new Widget());
if (map.Contains("EXAMPLE")) {
  ...
}
```

Should this consuming code change behavior based on some implementation detail of `GetMap`? 

Basically yes it doesn't matter if the dictionary is never mutated again but there is nothing stopping callers from doing so.

Reply:
Thanks for the explanation! I have fixed it.